ouroboros-consensus-cardano-test/test/Test/Consensus/Cardano/Translation.hs

```diff
@@ -0,0 +1,226 @@
+-- we'd be able to match against the similarly indexed TranslateEra values.
+shelleyToAllegraTranslateLedgerState ::
+  LedgerConfig (ShelleyBlock Proto (ShelleyEra Crypto))
+  -> LedgerConfig (ShelleyBlock Proto (AllegraEra Crypto))
+  -> TranslateLedgerState (ShelleyBlock Proto (ShelleyEra Crypto))
+       (ShelleyBlock Proto (AllegraEra Crypto))
+shelleyToAllegraTranslateLedgerState maryConfig allegraConfig =
+  provideBoth pair (WrapLedgerConfig maryConfig) (WrapLedgerConfig allegraConfig)
+  where
+    eraTranslations :: EraTranslation (CardanoEras Crypto)
+    eraTranslations = hardForkEraTranslation
+    PCons _ ps = translateLedgerState eraTranslations
+    PCons pair _ = ps
+
+translateLedgerStateShelleyToAllegra
+  :: LedgerConfig (ShelleyBlock Proto (ShelleyEra Crypto))
+  -> LedgerConfig (ShelleyBlock Proto (AllegraEra Crypto))
+  -> EpochNo
+  -> LedgerState (ShelleyBlock Proto (ShelleyEra Crypto)) EmptyMK
+  -> LedgerState (ShelleyBlock Proto (AllegraEra Crypto)) DiffMK
+translateLedgerStateShelleyToAllegra shelleyConfig allegraConfig epochNo shelleyLedgerState =
+  let translateLedgerState = shelleyToAllegraTranslateLedgerState shelleyConfig allegraConfig
+  in translateLedgerStateWith translateLedgerState epochNo shelleyLedgerState
+
+genShelleyLedgerState :: CanMock proto era => Gen (LedgerState (ShelleyBlock proto era) EmptyMK)
+genShelleyLedgerState = arbitrary
+
+run :: IO ()
+run = generate (extractShelleyLedgerTables <$> gen) >>= print
+  where
+    gen :: Gen (LedgerState (ShelleyBlock Proto (AllegraEra Crypto)) DiffMK)
+    gen = do
+      let shelleyC = fixedShelleyLedgerConfig
+      let allegraC = fixedShelleyLedgerConfig
+      state <- genShelleyLedgerState
+      pure $ translateLedgerStateShelleyToAllegra shelleyC allegraC epochNo state
+    epochNo :: EpochNo
+    epochNo = EpochNo 10
+
+translateToShelleyLedgerState
+  :: LedgerConfig ByronBlock
+  -> LedgerConfig (ShelleyBlock Proto Era)
+  -> EpochNo
+  -> LedgerState ByronBlock EmptyMK
+  -> LedgerState (ShelleyBlock Proto Era) DiffMK
+translateToShelleyLedgerState byronConfig shelleyConfig epochNo byronLedgerState =
+  let translateLedgerState = byronToShelleyTranslateLedgerState byronConfig shelleyConfig
+  in translateLedgerStateWith translateLedgerState epochNo byronLedgerState
+
+extractShelleyLedgerTables
+  :: LedgerState (ShelleyBlock Proto era) DiffMK
+  -> UtxoDiff (TxIn (EraCrypto era)) (Core.TxOut era)
+extractShelleyLedgerTables shelleyLedgerState =
+  let ApplyDiffMK tables = shelleyUTxOTable $ shelleyLedgerTables shelleyLedgerState
+  in tables
+
+fixedShelleyLedgerConfig :: Core.TranslationContext era ~ ()
+                         => ShelleyLedgerConfig era
+fixedShelleyLedgerConfig = mkShelleyLedgerConfig
+      shelleyGenesis
+      ()
+      (fixedEpochInfo (sgEpochLength shelleyGenesis) tpraosSlotLength)
+      (MaxMajorProtVer 1000)
+
+shelleyGenesis :: ShelleyGenesis era
```

Comment:
If we use a fixed Shelley genesis we should justify why. I think in this case the justification might be that we are interested in the initial UTxO set, which is not affected by the genesis config. But we need to confirm this.